docker-compose.yml

```diff
@@ -1,29 +1,56 @@
 version: '3.7'
 
+x-shared_environment: &shared_environment
+  LOG_LEVEL: ${LOG_LEVEL}
+  DATABASE_HOST: db
+  DATABASE_NAME: ${DATABASE_NAME:-vapor_database}
+  DATABASE_USERNAME: ${DATABASE_USERNAME:-vapor_username}
+  DATABASE_PASSWORD: ${DATABASE_PASSWORD:-vapor_password}
+
 services:
   {{name}}:
     image: {{name}}:latest
     build:
       context: .
+    environment:
+      <<: *shared_environment
```

Comment:
Would it be cleaner if we kept the `CMD` out of the dockerfile and instead passed it here?
```yml
cmd: serve --env production --hostname 0.0.0.0 --port 80
```

Reply:
One thing to note is that sometimes, depending on how CMD / ENTRYPOINT is configured, I've seen Docker not correctly send `^C` signals through to Vapor. If you have `--log trace` or `--log debug`, and you send `^C`, it should say "application shutting down".

Reply:
> Would it be cleaner if we kept the CMD out of the dockerfile and instead passed it here?
I like to keep `Dockerfile`s as standalone as possible. The Dockerfile describes an image that you could, for example, upload to Docker Hub and share with the world (it would not carry your docker-compose file along with it) -- it's nice if that image has a reasonable default built into it, even though the user of the image is welcome to replace the command themselves as well (whether in a command line argument or in their own docker-compose file).

Reply:
> One thing to note is that sometimes...

I'll look into this.

Reply:
Fixed!